WordPress/Classes/ViewRelated/Plugins/CollectionViewContainerRow.swift

```diff
@@ -0,0 +1,178 @@
+        return dataSourceItems.count
+    }
+
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: CellReuseIdentifier, for: indexPath)
+
+        let dataItem = dataSourceItems[indexPath.item]
+        configureCollectionCellBlock(cell, dataItem)
+
+        return cell
+    }
+
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        collectionCellSeletectedBlock(dataSourceItems[indexPath.item])
+    }
+
+}
+
+class CollectionViewContainerCell: UITableViewCell {
+
+    private(set) var collectionView: UICollectionView!
+    private(set) var titleLabel: UILabel!
+    private(set) var actionButton: UIButton!
+
+    var buttonTappedAction: (() -> Void)?
+
+    override init(style: UITableViewCellStyle, reuseIdentifier: String?) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+        setupViews()
+    }
+
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private func setupViews() {
+        selectionStyle = .none
+
+        titleLabel = UILabel(frame: .zero)
+        titleLabel.font = UIFont.systemFont(ofSize: UIFont.preferredFont(forTextStyle: .title2).pointSize,
+                                            weight: .bold)
+        titleLabel.textColor = WPStyleGuide.darkGrey()
+        titleLabel.translatesAutoresizingMaskIntoConstraints = false
+
+        self.addSubview(titleLabel)
+
+        titleLabel.topAnchor.constraint(equalTo: self.topAnchor, constant: 18).isActive = true
+        titleLabel.leadingAnchor.constraint(equalTo: self.leadingAnchor, constant: 18).isActive = true
+
+        actionButton = UIButton(type: .custom)
+        actionButton.setTitleColor(WPStyleGuide.mediumBlue(), for: .normal)
+        actionButton.addTarget(self, action: #selector(actionButtonTapped), for: .touchUpInside)
+        actionButton.titleLabel?.font = UIFont.preferredFont(forTextStyle: .body)
+        actionButton.translatesAutoresizingMaskIntoConstraints = false
+
+        self.addSubview(actionButton)
+
+        actionButton.trailingAnchor.constraint(equalTo: self.trailingAnchor, constant: -18).isActive = true
+        actionButton.lastBaselineAnchor.constraint(equalTo: titleLabel.lastBaselineAnchor).isActive = true
+        actionButton.setContentHuggingPriority(.required, for: .horizontal)
+
+        titleLabel.trailingAnchor.constraint(greaterThanOrEqualTo: actionButton.leadingAnchor).isActive = true
+
+        let flowLayout = UICollectionViewFlowLayout()
+        flowLayout.scrollDirection = .horizontal
+        flowLayout.itemSize = CGSize(width: 98, height: 98*2)
+        flowLayout.minimumLineSpacing = 18
+        flowLayout.sectionInset = UIEdgeInsets(top: 0, left: 18, bottom: 0, right: 18)
```

Comment:
Probably a good idea to move the magic numbers over to a Constants structure maybe?

```
struct Constants {
   static let itemSize = ...
}
```